### PR TITLE
Add debugger engine version checker

### PR DESCRIPTION
**Pull request by knightburton**
This patch add a new script to the tools.
The new tool is clones the iotjs and the jerry (based on the submodule hash)
into a temporary folder then compare the jerry debugger version number with
the iotjscode debugger version number.

IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

**Comment by knightburton** [comment]
Note: The `npm install nodegit --build-from-source` is need to be able to use the nodegit correctly, because of the travis a bit outdated. This has a huge impact on the build time and not a nice workaround, I am searching for an other (maybe smaller) node package to access to the git.

File: tools/check-debugger-version.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    .then(repo => repo.getHeadCommit())
+    .then(commit => commit.getEntry('jerry-core/debugger/debugger.h'))
+    .then(entry => entry.getBlob())
+    .then(blob => blob.toString())
+    .then(data => data.split('\n')
+      .find(line => line.includes('#define JERRY_DEBUGGER_VERSION'))
+      .match(/^#.+\(([0-9])\)$/)[1])
+    .then(version => {
+      if (+version === JERRY_DEBUGGER_VERSION) {
+        console.info('The IoT.js Code and IoT.js debugger version is match.');
```
Comment (rtakacs) [inline tools/check-debugger-version.js]: ... are match.

File: tools/check-debugger-version.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        .then(() => submodule.update(1))
+        .then(() => submodule.open());
+    })
+    .then(repo => repo.getHeadCommit())
+    .then(commit => commit.getEntry('jerry-core/debugger/debugger.h'))
+    .then(entry => entry.getBlob())
+    .then(blob => blob.toString())
+    .then(data => data.split('\n')
+      .find(line => line.includes('#define JERRY_DEBUGGER_VERSION'))
+      .match(/^#.+\(([0-9])\)$/)[1])
```
Comment (rtakacs) [inline tools/check-debugger-version.js]: I think it doesn't need to split the input string. The following code also should work (not tested):
```js
.then(data => data.match(/#define JERRY_DEBUGGER_VERSION\s+(\d+)/m)[1])
```

File: tools/check-debugger-version.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    .then(version => {
+      if (+version === JERRY_DEBUGGER_VERSION) {
```
Comment (rtakacs) [inline tools/check-debugger-version.js]: I think the equality operator (==) should be enough. (It does type conversions.)

Comment (knightburton) [reply tools/check-debugger-version.js]: Yes, should be, however in javascript, using a Strict Equality Comparison is highly recommended so I would not change that.

File: tools/check-debugger-version.js (hunk 2 above)
Comment (knightburton) [reply tools/check-debugger-version.js]: I tested that and I think that version is much slower than working on an array of lines.

**Comment by knightburton** [comment]
@rtakacs I've updated the PR.

File: tools/check-debugger-version.js (hunk 2 above)
Comment (rtakacs) [reply tools/check-debugger-version.js]: You are right, the code that I wrote is slower.

**Comment by rtakacs** [comment]
LGTM

File: tools/check-debugger-version.js (hunk 3 above)
Comment (rtakacs) [reply tools/check-debugger-version.js]: Okay, the change isn't necessary.

**Comment by yichoi** [comment]
@knightburton please rebase it to the master

**Comment by knightburton** [comment]
@yichoi I've rebased the PR.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**